<launch>
  <arg name="image_id" default="0"/>
  <arg name="use_real_camera" default="false" />

  <node pkg="simod_rolls_detection" type="close_line_detection_node" name="close_line_detection_node" output="screen">

    <param name="rgb_image_topic" type="string" value="/oak/rgb/image_raw" />
    <param name="camera_info_topic" type="string" value="/oak/rgb/camera_info" />

    <!-- increase this if first camera frames are bad for some reason -->
    <param name="discard_first_camera_frames" type="int" value="0" />

    <param name="mode_basic" type="bool" value="false" />

    <param name="roll_diameter" type="double" value="0.11" />
  </node>

  <node pkg="simod_rolls_detection" type="close_line_detection_test" name="close_line_detection_test" output="screen">

    <param name="world_frame_id" type="string" value="map" />
    <param name="camera_frame_id" type="string" value="camera" />

    <param name="rgb_image_topic" type="string" value="/oak/rgb/image_raw" />
    <param name="camera_info_topic" type="string" value="/oak/rgb/camera_info" />

    <param name="use_real_camera" type="bool" value="$(arg use_real_camera)" />

    <param name="rgb_filename" type="string" value="$(find simod_rolls_detection)/data/bologna_29-07-25/Test Camera 29-07-25 Superiori Inclinate/image$(arg image_id).png" />
    <param name="camera_info_filename" type="string" value="$(find simod_rolls_detection)/data/bologna_29-07-25/Test Camera 29-07-25 Superiori Inclinate/camerainfo$(arg image_id).txt" />
    <param name="camera_pose_filename" type="string" value="$(find simod_rolls_detection)/data/bologna_29-07-25/Test Camera 29-07-25 Superiori Inclinate/pose$(arg image_id).txt" />
    <param name="layer_height" type="double" value="0.0" />      <!-- z of the top layer in world coordinates -->
    <param name="initial_guess_y" type="double" value="0.8" /> <!-- initial guess y of the line in world coordinates -->
    <param name="initial_guess_y" type="double" value="0.85" if="$(eval image_id == 3 or image_id == 4 or image_id == 5)" />
    <param name="initial_guess_y" type="double" value="0.9" if="$(eval image_id == 0 or image_id == 1 or image_id == 11)" />
    <param name="initial_guess_y" type="double" value="1.1" if="$(eval image_id == 2 or image_id == 10)" />
   
    <param name="initial_guess_x" type="double" value="0.0" />
    <param name="initial_guess_window_size_x" type="double" value="1.0" />
    <param name="initial_guess_window_size_y" type="double" value="0.2" />
  </node>

  <include file="$(find simod_camera)/launch/oak_d_pro.launch" if="$(arg use_real_camera)" />
</launch>
